src/Angor/Client/Pages/Signatures.razor

```diff
@@ -357,26 +437,117 @@
                 sigJson);
 
             FounderProject.LastRequestForSignaturesTime = SignService.SendSignaturesToInvestor(encryptedContent, nostrPrivateKeyHex, signature.investorPubKey, signature.EventId);
-
             Storage.UpdateFounderProject(FounderProject);
 
             signaturesRequests.Single(_ => _.investorPubKey == signature.investorPubKey && _.TimeApproved is null)
                 .TimeApproved = FounderProject.LastRequestForSignaturesTime;
 
             return new OperationResult { Success = true };
-        });
+        }
+        catch (Exception ex)
+        {
+            return new OperationResult { Success = false, Message = $"An error occurred: {ex.Message}" };
+        }
+    }
 
-        if (operationResult.Success)
+
+
+
+
+
+    private async Task ApproveSignature(SignatureRequest signature)
+    {
+        var words = await passwordComponent.GetWalletAsync();
```

Comment:
note that the method `GetWalletAsync` can't return a null (it will throw if words are not found).
So no need to check they are not null.